shaders/BooleanInverseColorShader.shader

```diff
@@ -0,0 +1,13 @@
+shader_type canvas_item;
+
+uniform vec4 fg_col : hint_color = vec4(0.745098, 0.356863, 0.921569, 1);
+uniform vec4 bg_col : hint_color = vec4(0.290196, 0.14902, 0.360784, 1);
+void fragment() {
+    vec4 screen_col = textureLod(SCREEN_TEXTURE, SCREEN_UV, 0.0);
+    float alpha = texture(TEXTURE, UV).a;
+    if ((fg_col - screen_col).x < 0.1) {
+        COLOR = vec4(bg_col.xyz, alpha);
+    } else {
+        COLOR = vec4(fg_col.xyz, alpha);
+    }
+}
```

Comment:
Add a blank line here.

I think this file should be renamed, as right now this seems to not in fact inverse the colour, instead this selects one of two colours based on how far they are from what the current colour is.
Also has this been tested with gles2 mode? A basic GUI part should be tested to work fine there as well.

Reply:
Gles2 mode works fine with this in my laptop

Reply:
That's good, I remember there being some post somewhere, that (maybe it was older version) Godot gles2 shaders were not allowed to read the screen, or maybe it was that it causes just a performance impact now?

Reply:
If there is a major performance impact (more than like 5 FPS at around 60 FPS), I think we need a performance option to not use the shader.

Reply:
I don't see any noticeable decrease in fps tbh (at least for me), maybe someone else could also test it out

Reply:
I think that's good enough, I may be just remembering some old posts that screen reading shaders were really bad in GLES2, we can add an option later if needed to disable the effect.